interfaces/repo_test.go

```diff
@@ -848,3 +848,179 @@ func (s *RepositorySuite) TestSecuritySnippetsForSnapFailureWithPermanentSnippet
 	c.Assert(err, ErrorMatches, "cannot compute static snippet for consumer")
 	c.Check(snippets, IsNil)
 }
+
+// Tests for AddSnap and RemoveSnap
+
+type AddRemoveSuite struct {
+	repo *Repository
+}
+
+var _ = Suite(&AddRemoveSuite{})
+
+func (s *AddRemoveSuite) SetUpTest(c *C) {
+	s.repo = NewRepository()
+	err := s.repo.AddInterface(&TestInterface{InterfaceName: "iface"})
+	c.Assert(err, IsNil)
+	err = s.repo.AddInterface(&TestInterface{
+		InterfaceName:        "invalid",
+		SanitizePlugCallback: func(plug *Plug) error { return fmt.Errorf("plug is invalid") },
+		SanitizeSlotCallback: func(slot *Slot) error { return fmt.Errorf("slot is invalid") },
+	})
+	c.Assert(err, IsNil)
+}
+
+const testConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [iface]
+`
+const testDifferentIfaceConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [iface]
+plugs:
+    iface:
+        interface: other-iface
+`
+const testDifferentAttrsConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [iface]
+plugs:
+    iface:
+        attr: extra
+`
+const testUnknownIfaceConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [unknown]
+`
+const testInvalidIfaceConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [invalid]
+`
+const testDifferentLabelConsumerYaml = `
+name: consumer
+apps:
+    app:
+        plugs: [iface]
+plugs:
+    iface:
+        label: New Label
+`
+const testBareConsumerYaml = `
+name: consumer
+apps:
+    app:
+`
+
+const testProducerYaml = `
+name: producer
+apps:
+    app:
+        slots: [iface]
+`
+const testDifferentIfaceProducerYaml = `
```

Comment:
There's a TODO below, at the end of all tests to clone the tests for Slot side of the code. That should use all of those. The rest should be indeed removed. I should probably just do that.